WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.java

```diff
@@ -44,7 +46,16 @@ public abstract class BlockProcessor {
     BlockProcessor(String localId, MediaFile mediaFile) {
         mLocalId = localId;
         mRemoteId = mediaFile.getMediaId();
-        mRemoteUrl = org.wordpress.android.util.StringUtils.notNullStr(Utils.escapeQuotes(mediaFile.getFileURL()));
+
+        String mediaURL;
+        if (!TextUtils.isEmpty(mediaFile.getFileUrlLargeSize())) {
+            mediaURL = mediaFile.getFileUrlLargeSize();
+        } else if (!TextUtils.isEmpty(mediaFile.getFileUrlMediumSize())) {
+            mediaURL = mediaFile.getFileUrlMediumSize();
+        } else {
+            mediaURL = mediaFile.getFileURL();
+        }
```

Comment:
I wonder if we can extract this to a method on the `MediaFile`, e.g. something like `mediaFile.getOptimalFileURL`, since we use the same logic [here](https://github.com/wordpress-mobile/WordPress-Android/pull/16875/files#diff-3d055f27ee69cf53a3055662daab1fc04df6e0eba7deadfd63c3a79df8e96dc5R1346). Wdyt?

Reply:
> I've updated it from your comments, let me know if you can give it another check!

Thanks for updating it! I've left a couple more minor comments that maybe we could simplify it even further. Thanks for your work on this!